tests/test_server/test_endpoints.py

```diff
@@ -0,0 +1,200 @@
+        _add_pipeline_endpoint(
+            app,
+            endpoint_config=Mock(endpoint="/predict/parse_int"),
+            pipeline=Mock(input_schema=StrSchema, output_schema=int, side_effect=parse),
+        )
+        assert app.routes[-1].path == "/predict/parse_int"
+        assert app.routes[-1].response_model is int
+        assert app.routes[-1].methods == {"POST"}
+
+        for v in ["1234", "5678"]:
+            response = client.post("/predict/parse_int", json=dict(value=v))
+            assert response.status_code == 200
+            assert response.json() == int(v)
+
+    def test_add_model_endpoint_with_from_files(self, app):
+        _add_pipeline_endpoint(
+            app,
+            endpoint_config=Mock(endpoint="/predict/parse_int"),
+            pipeline=Mock(input_schema=FromFilesSchema, output_schema=int),
+        )
+        assert app.routes[-1].path == "/predict/parse_int/files"
+        assert app.routes[-1].response_model is int
+        assert app.routes[-1].methods == {"POST"}
+
+    def test_add_invocations_endpoint(self, app):
+        _add_pipeline_endpoint(
+            app,
+            endpoint_config=Mock(endpoint="/predict/parse_int"),
+            pipeline=Mock(input_schema=FromFilesSchema, output_schema=int),
+            add_invocations_endpoint=True,
+        )
+        assert app.routes[-1].path == "/invocations"
+
+    def test_add_endpoint_with_no_route_specified(self, app):
+        _add_pipeline_endpoint(
+            app,
+            endpoint_config=Mock(endpoint=None),
+            pipeline=Mock(input_schema=StrSchema, output_schema=int),
+        )
+        assert app.routes[-1].path == "/predict"
+
+
+class TestActualModelEndpoints:
+    @pytest.fixture(scope="class")
+    def client(self):
+        server_config = ServerConfig(
+            num_cores=1,
+            num_workers=1,
+            endpoints=[
+                EndpointConfig(
+                    name="test endpoint 1",
+                    endpoint="/predict/static-batch",
+                    task="text-classification",
+                    model=Pipeline.default_model_for("text-classification"),
+                    batch_size=2,
+                ),
+                # TODO add these back in
+                # EndpointConfig(
+                #     name="test endpoint 2",
+                #     endpoint="/predict/dynamic-batch",
+                #     task="text-classification",
+                #     model=Pipeline.default_model_for("text-classification"),
+                #     batch_size=1,
+                # ),
+                # EndpointConfig(
+                #     name="test endpoint 3",
+                #     endpoint="/predict/bucketed",
+                #     task="text-classification",
+                #     model=Pipeline.default_model_for("text-classification"),
+                #     batch_size=1,
+                #     bucketing=SequenceLengthsConfig(sequence_lengths=[2, 4]),
+                # ),
```

Comment:
Remove or add back in?

Reply:
Oh yeah thanks for reminding me, I wanted to add more tests here. Ideally we could get the mock config PR before this so they aren't super slow.